File: app/src/main/java/droiddevelopers254/droidconke/viewmodels/EventTypeViewModel.java
package droiddevelopers254.droidconke.viewmodels;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MediatorLiveData;
import android.arch.lifecycle.ViewModel;

import droiddevelopers254.droidconke.datastates.EventTypeState;
import droiddevelopers254.droidconke.models.EventTypeModel;
import droiddevelopers254.droidconke.models.SessionsModel;
import droiddevelopers254.droidconke.models.WifiDetailsModel;
import droiddevelopers254.droidconke.repository.EventTypeRepo;
import droiddevelopers254.droidconke.repository.WifiDetailsRepo;

public class EventTypeViewModel extends ViewModel {
    private MediatorLiveData<EventTypeState> eventTypeModelMediatorLiveData;
    private EventTypeRepo eventTypeRepo;
    private MediatorLiveData<WifiDetailsModel> wifiDetailsModelMediatorLiveData;
    private WifiDetailsRepo wifiDetailsRepo;

    public EventTypeViewModel(){
        eventTypeModelMediatorLiveData= new MediatorLiveData<>();
        eventTypeRepo= new EventTypeRepo();
        wifiDetailsModelMediatorLiveData= new MediatorLiveData<>();
        wifiDetailsRepo= new WifiDetailsRepo();
    }
    public LiveData<EventTypeState> getSessions(){
        return eventTypeModelMediatorLiveData;
    }

    public LiveData<WifiDetailsModel> getWifiDetails(){
        return wifiDetailsModelMediatorLiveData;
    }

    public void fetchSessions(){
        final LiveData<EventTypeState> eventTypeModelLiveData=eventTypeRepo.getSessionData();
        eventTypeModelMediatorLiveData.addSource(eventTypeModelLiveData,
                sessionsModelMediatorLiveData -> {
            if (this.eventTypeModelMediatorLiveData.hasActiveObservers()){
                this.eventTypeModelMediatorLiveData.removeSource(eventTypeModelLiveData);
            }
            this.eventTypeModelMediatorLiveData.setValue(sessionsModelMediatorLiveData);
                });
    }
    public void fetchRemoteConfigValues(){
        final LiveData<WifiDetailsModel>wifiDetailsModelLiveData=wifiDetailsRepo.getWifiDetails();
        wifiDetailsModelMediatorLiveData.addSource(wifiDetailsModelLiveData,
                wifiDetailsModelMediatorLiveData->{
            if (this.wifiDetailsModelMediatorLiveData.hasActiveObservers()){
                this.wifiDetailsModelMediatorLiveData.removeSource(wifiDetailsModelLiveData);
            }
            this.wifiDetailsModelMediatorLiveData.setValue(wifiDetailsModelMediatorLiveData);
                });
    }
}

File: app/src/main/java/droiddevelopers254/droidconke/viewmodels/EventTypeViewModel.kt
package droiddevelopers254.droidconke.viewmodels

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import droiddevelopers254.droidconke.datastates.Result
import droiddevelopers254.droidconke.models.EventTypeModel
import droiddevelopers254.droidconke.repository.EventTypeRepo
import kotlinx.coroutines.launch

class EventTypeViewModel(private val eventTypeRepo: EventTypeRepo) : ViewModel() {
  private val eventTypeModelMediatorLiveData = MutableLiveData<List<EventTypeModel>>()
  private val eventDetailsError = MutableLiveData<String>()


  fun getWifiDetailsResponse(): LiveData<List<EventTypeModel>> = eventTypeModelMediatorLiveData

  fun getWifiDetailsError(): LiveData<String> = eventDetailsError


  fun fetchSessions() {
    viewModelScope.launch {
      when (val value = eventTypeRepo.getSessionData()) {
          is Result.Success -> eventTypeModelMediatorLiveData.postValue(value.data)
          is Result.Error -> eventDetailsError.postValue(value.exception)
      }
    }

  }
}
